apps/provider-console/src/components/become-provider/ProviderPricing.tsx

```diff
@@ -0,0 +1,336 @@
+"use client";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { Button, Form, FormControl, FormDescription, FormField, FormItem, FormLabel, Input, Separator, Slider } from "@akashnetwork/ui/components";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useAtom } from "jotai";
+import { ChevronDownIcon } from "lucide-react";
+import { z } from "zod";
+
+import providerProcessStore from "@src/store/providerProcessStore";
+import { ResetProviderForm } from "./ResetProviderProcess";
+
+interface ProviderPricingProps {
+  resources?: {
+    cpu: number;
+    memory: number;
+    storage: string;
+    persistentStorage: number;
+    gpu: number;
+  };
+  stepChange: () => void;
+}
+
+const providerPricingSchema = z.object({
+  cpu: z.number().min(0),
+  memory: z.number().min(0),
+  storage: z.number().min(0),
+  gpu: z.number().min(0),
+  persistentStorage: z.number().min(0),
+  ipScalePrice: z.number().min(0),
+  endpointBidPrice: z.number().min(0)
+});
+
+type ProviderPricingValues = z.infer<typeof providerPricingSchema>;
+
+export const ProviderPricing: React.FC<ProviderPricingProps> = ({ stepChange }) => {
+  const [providerProcess, setProviderProcess] = useAtom(providerProcessStore.providerProcessAtom);
+  const [resources, setResources] = useState({
+    cpu: 24,
+    memory: 724,
+    storage: 1024,
+    persistentStorage: 7024,
+    gpu: 5
+  });
+
+  useEffect(() => {
+    const calculateResources = () => {
+      let totalCpu = 0;
+      let totalMemory = 0;
+      let totalStorage = 0;
+      let totalPersistentStorage = 0;
+      let totalGpu = 0;
+
+      providerProcess.machines.forEach(machine => {
+        totalCpu += parseInt(machine.systemInfo.cpus, 10);
+        totalMemory += parseInt(machine.systemInfo.memory.replace("Gi", ""), 10);
+        machine.systemInfo.storage.forEach((storage, index) => {
+          if (index === 0) {
+            totalStorage += storage.size / (1024 * 1024 * 1024);
+          } else {
+            totalPersistentStorage += storage.size / (1024 * 1024 * 1024);
+          }
+        });
+        totalGpu += machine.systemInfo.gpu.count;
+      });
+
+      setResources({
+        cpu: totalCpu,
+        memory: totalMemory,
+        storage: totalStorage,
+        persistentStorage: totalPersistentStorage,
+        gpu: totalGpu
+      });
+    };
+
+    calculateResources();
+  }, [providerProcess.machines]);
+
+  const [showAdvanced, setShowAdvanced] = useState(false);
+  const form = useForm<ProviderPricingValues>({
+    resolver: zodResolver(providerPricingSchema),
+    defaultValues: {
+      cpu: 1.6,
+      memory: 0.8,
+      storage: 0.02,
+      gpu: 100,
+      persistentStorage: 0.3,
+      ipScalePrice: 5,
+      endpointBidPrice: 0.5
+    }
+  });
+
+  const watchValues = form.watch();
+
+  const calculateEstimatedEarnings = (values: ProviderPricingValues) => {
```

Comment:
such functions are generally better be wrapped into `useCallback`

Reply:
change to useCallBcak